pulsar-common/src/main/proto/PulsarApi.proto

```diff
@@ -272,6 +289,8 @@ message CommandProducer {
 
     /// Add optional metadata key=value to this producer
     repeated KeyValue metadata    = 6;
+
+	optional int64 schema_version = 7;
```

Comment:
I think that when creating producer & consumer we need to specify the actual schema so that we can validate on broker side, and if that fails, the producer creation fails.

Reply:
This seems to be an oversight on my part. Since this series of changes is concerned with a 'repository' I appear to have glossed over this point.

Reply:
Sure, no problem. I think it would be good to have the "final" version of the `PulsarApi.proto` to get a sense of all the changes in the interaction between client & broker.

Reply:
Do we need to specify the entire schema, or just the name and the version?

Reply:
It should carry the whole schema here. The application/producer doesn't necessarely know what the version is. 

Eg: if I'm using a type like : 

```java
class MyClass {
   public int a;
   public int b;
};
```

When I try to publish, I just know the format I'm about to send, and we can extract the avro/json schema out of it. When I try to create producer, I can send that information, and the broker can compare with the schemas already availalable for this topic. If the schema is know, we just return the schema version in the `ProducerSuccess`  response. If it's a new (compatible) version of the schema, it will also be stored.